src/sentry/api/endpoints/project_searches.py

```diff
@@ -17,6 +17,8 @@ class SavedSearchSerializer(serializers.Serializer):
 
 
 class ProjectSearchesEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectSearchPermission,)
```

Comment:
@dcramer As the issue stated "members can't add new saved search", I tried to resolve it by adding new permission keeping in mind that a member would be allowed to add and change his default search view. Also API should be robust enough to prevent deleting of search view by members.
Existing permissions shouldn't mix with search permissions as both serve different purpose thats the reason I created separate permissions.

If still we don't require separate permissions, I will remove that and simply hide the button in UI depending on existing permissions.